tutorials/todo/src/main.sw

```diff
@@ -0,0 +1,51 @@
+contract;
+
+dep abi;
+dep tasks;
+
+use std::storage::StorageMap;
+use abi::Todo;
+use tasks::Task;
+
+storage {
+    tasks: StorageMap<u64, Task>,
+}
+
+// TODO: remove hack array once mutable arrays are ready see https://github.com/FuelLabs/sway/issues/1626
+
+impl Todo for Contract {
+     fn add_task(index: u64 value: str[20]) {
+        // logic needs to be fixed to make sure they can only add up to 5 tasks
```

Comment:
You can keep track of the total number of tasks in storage.
Take a look at the drafts for the multisig, escrow and fundraiser on use of `require` statements to handle error handling logic